src/app/ecosystem-explorer/project-form/components/EcosystemProjectForm.tsx

```diff
@@ -0,0 +1,225 @@
+import { FormField } from '@/components/Form/FormField'
+
+import {
+  BRIEF_CHARACTER_LIMIT,
+  MAX_FILE_SIZE_IN_BYTES,
+  NETWORK_USE_CASE_CHARACTER_LIMIT,
+} from '../constants'
+import { useSubmitEcosystemProjectForm } from '../hooks/useSubmitEcosystemProjectForm'
+import { EcosystemProjectFormSchema } from '../schema/form'
+import { getOptionsFromObject } from '../utils/getOptionsFromObject'
+import { getYearOptions } from '../utils/getYearOptions'
+
+import { FormSection } from './FormSection'
+
+type StringOrUndefined = string | undefined
+
+export type EcosystemProjectFormData = z.infer<
+  typeof EcosystemProjectFormSchema
+>
+
+type ProjectFormProps = {
+  categoryData: CategoryMap
+  subCategoryData: CategoryMap
+  initialValues: EcosystemProjectFormData
+}
+
+export function EcosystemProjectForm({
+  categoryData,
+  subCategoryData,
+  initialValues,
+}: ProjectFormProps) {
+  const yearOptions = getYearOptions('desc')
+  const categoryOptions = getOptionsFromObject(categoryData)
+  const subCategoryOptions = getOptionsFromObject(subCategoryData)
+
+  const form = useForm<EcosystemProjectFormData>({
+    resolver: zodResolver(EcosystemProjectFormSchema),
+    defaultValues: initialValues,
+  })
+  const isSubmitting = form.formState.isSubmitting
+
+  const submitForm = useSubmitEcosystemProjectForm()
+
+  return (
+    <ControlledForm<EcosystemProjectFormData>
+      form={form}
+      className="max-w-readable"
+      onSubmit={submitForm}
+    >
+      <FormSection
+        title="Personal Information"
+        description="We'll use this to verify your site updates and reach out to you in case of questions."
+      >
+        <ControlledFormInput<EcosystemProjectFormData>
+          name="name"
+          label="Name"
+          placeholder="Your Name"
+          disabled={isSubmitting}
+        />
+        <ControlledFormInput<EcosystemProjectFormData>
+          name="email"
+          label="Email"
+          placeholder="Your Email"
+          type="email"
+          disabled={isSubmitting}
+        />
+      </FormSection>
+
+      <FormSection title="Project Details">
+        <ControlledFormInput<EcosystemProjectFormData>
+          name="projectName"
+          label="Project Name"
+          placeholder="Project Name"
+          disabled={isSubmitting}
+        />
+
+        <FormField
+          label="Which technology does your project utilize?"
+          error={form.formState.errors.tech?.root?.message}
+        >
+          <div className="flex flex-col gap-4">
+            <ControlledFormCheckbox<EcosystemProjectFormData>
+              name="tech.filecoin"
+              label="Filecoin"
+              disabled={isSubmitting}
+            />
+            <ControlledFormCheckbox<EcosystemProjectFormData>
+              name="tech.ipfs"
+              label="IPFS"
+              disabled={isSubmitting}
+            />
+          </div>
+        </FormField>
```

Comment:
We get this error in the browser - I manage to isolate it to this part, tho I don't know what is causing it.

<img width="1266" alt="Screenshot 2024-09-12 at 14 39 44" src="https://github.com/user-attachments/assets/0ab2726d-86f8-4d5b-b5c4-a1173d76c944">

Reply:
Should be fixed.